misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/reporting/translating/TranslationSupplierBasedTranslatorSpec.kt

```diff
@@ -4,30 +4,33 @@ import ch.tutteli.atrium.api.fluent.en_GB.toBe
 import ch.tutteli.atrium.api.verbs.internal.expect
 import ch.tutteli.atrium.core.coreFactory
 import ch.tutteli.atrium.reporting.translating.*
-import ch.tutteli.atrium.specs.describeFun
-import com.nhaarman.mockitokotlin2.doReturn
-import com.nhaarman.mockitokotlin2.mock
-import org.jetbrains.spek.api.Spek
-import org.jetbrains.spek.api.dsl.SpecBody
-import org.jetbrains.spek.api.dsl.context
-import org.jetbrains.spek.api.dsl.describe
-import org.jetbrains.spek.api.dsl.it
-
-//TODO #116 migrate spek1 to spek2 - move to specs-common
+import ch.tutteli.atrium.specs.describeFunTemplate
+import io.mockk.*
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.Suite
+import org.spekframework.spek2.style.specification.describe
+import org.spekframework.spek2.dsl.ScopeBody
+
 abstract class TranslationSupplierBasedTranslatorSpec(
     testeeFactory: (translationSupplier: TranslationSupplier, localeOrderDecider: LocaleOrderDecider, locale: Locale, fallbackLocals: List<Locale>) -> Translator,
     describePrefix: String = "[Atrium] "
 ) : Spek({
 
-    fun describeFun(vararg funName: String, body: SpecBody.() -> Unit) =
-        describeFun(describePrefix, funName, body = body)
+    fun describeFun(vararg funName: String, body: Suite.() -> Unit) =
+        describeFunTemplate(describePrefix, funName, body = body)
 
     fun testeeFactory(translationSupplier: TranslationSupplier, locale: Locale, vararg fallbackLocals: Locale) =
         testeeFactory(translationSupplier, coreFactory.newLocaleOrderDecider(), locale, fallbackLocals.toList())
 
-    fun mockTranslationProvider(locale: Locale, translatable: Translatable, translation: String): TranslationSupplier {
-        return mock {
-            on { get(translatable, locale) }.doReturn(translation)
+    fun mockTranslationProvider(locale: Locale, translatable: Translatable, translation: String?): TranslationSupplier {
+        val localeSlot = slot<Locale>()
+        val translatableSlot = slot<Translatable>()
+        return mockk {
+            every { get(capture(translatableSlot), capture(localeSlot)) } answers {
+                // println("translatableSlot.captured.value:" + (translatableSlot.captured as StringBasedTranslatable).value);
+                // println("translatable.value:" + (translatable as StringBasedTranslatable).value);
+                if ((localeSlot.captured != locale) || ((translatableSlot.captured as StringBasedTranslatable).value != (translatable as StringBasedTranslatable).value)) null else translation 
```

Comment:
that's a change in behaviour. Could you briefly explain why?

Reply:
Also related to strictness of the framework
There is no implementation for a single call to _get_ method which was the requirement here
So I had to come up with a way of separating the single scenario required in this scope which is calling _get_ with specific parameters

Hope it explains my point better, if you would like we can have discuss about it

Reply:
I see the problem of the strictness and I obviously did not have a good look at your implementation, it is still the same behaviour but I think written in a complicated way. I think the following would be easier to understand:
```
return mockk {
  every { get(any(), any()) } returns null
  every { get(translatable, locale) } returns translation
}
```